docs/DeveloperGuide.md

```diff
@@ -1,34 +1,284 @@
+- [4. Appendix I: Requirements](#4-appendix-i-requirements)
+  * [4.1 Product scope](#41-product-scope)
+    + [4.1.1 Target user profile](#411-target-user-profile)
+    + [4.1.2 Value](#412-value)
+  * [4.2. User Stories](#42-user-stories)
+  * [4.3. Non-Functional Requirements](#43-non-functional-requirements)
+  * [4.4. Glossary](#44-glossary)
+- [5. Appendix II: Instructions for manual testing](#5-appendix-ii-instructions-for-manual-testing)
+
+## 1. Setting up, getting started
+
+Refer to the guide [Setting up and getting started](https://github.com/AY2021S1-CS2113T-F14-3/tp/blob/master/README.md#duke-project-template).
+
+## 2. Design
+
+This section describes the design and implementation of the product. It has been divided into two sections: Architecture 
+and Implementation. 
+
+### 2.1. Architecture
+
+The architecture diagram given in the figure below explains the high-level design of the App. 
+
+![Architecture Diagram](DG_Diagrams/Architecture.png)
+
+Given below is a quick overview of each component.
+
+`Main` is the class Duke.java. It is responsible for:
+
+- Initializing the components in the correct sequence, and calling or creating objects in the order of execution.
+
+The rest of the App consists of 4 main components:
+
+- `Ui` : The Ui component handles all interactions with the user.
+- `Logic` : The Logic component makes sense of the command and executes it.
+- `Model` : The Model component is responsible for all data held in the memory.
+- `Storage` : The Storage component handles data by reading from and writing to files in the hard disk. 
+
+The sections below give more details of each component.
+
+#### 2.1.1 Ui Component
+The Ui component is responsible for getting the user input and for displaying appropriate messages as response 
+to commands executed by the logic component. The following class diagram gives a brief overview of the Ui component 
+and it's interaction with other components.
+
+![uicomponent](DG_Diagrams/UiComponent.png)
+
+#### 2.1.2. Logic Component
+The Logic component is responsible for the following tasks:
+- Makes sense of the command entered by the user by splitting it into the command type and the parameters and/or 
+delimiters if any.
+- Executes a command based on its type and provides an appropriate result to the user via the Ui component.
+
+The following class diagram briefly explains how different classes in the Logic component interact with each other.
+
+<img src="DG_Diagrams/LogicComponent.png" alt="logiccomponent" width=1100 height=500>
+
+#### 2.1.3. Model Component
+The Model component is responsible for the following tasks:
+- Stores the bus data, dine info data and favourites list in memory.
+- Performs operations on the data to facilitate the command execution by Logic component.
+
+The following class diagram briefly explains how different classes in the Model component interact with each other.
+
+![modelcomponent](DG_Diagrams/ModelComponent.png)
```

Comment:
Perhaps it might be good to explain what headers `C` and `E` means in the case of diagrams?